bigtop-packages/src/common/apex/do-component-build

```diff
@@ -28,6 +28,9 @@ command -v mvn > /dev/null 2>&1 || {
 # Build apex
 mvn clean package -DskipTests "$@"
 
+#Find hadoop.version from pom.xml
+APEX_HADOOP_VERSION=$(grep -oPm1 "(?<=<hadoop.version>)[^<]+" pom.xml)
```

Comment:
@rsv that sums it up. Indeed it should be possible to pass the hadoop.version to the apex build.

I would also like to point out that the Apex package can be assembled without building Apex from source, by optionally using the official release artifacts from central:  https://github.com/atrato/apex-cli-package/blob/master/pom.xml

That script also shows how the somewhat quirky dependency filtering pom can be simplified (and the stray legacy datatorrent references removed).